### PR TITLE
Add CPA emails

### PR DESCRIPTION
**Pull request by carl-codeorg**
Adds the three email templates for CPA along with the ParentMailer methods that will be used to send the emails, once other CPA work is done.

## Links

[Parent Letter Spec](https://docs.google.com/document/d/1Ml15JK6mcZBHgiBRc-Cpj11yZ53Gu8gD2tnNLVwfzGM/edit)
[JIRA Ticket](https://codedotorg.atlassian.net/browse/P20-134)

## Testing story

In addition to the unit tests, previews of the emails can be viewed locally at http://localhost-studio.code.org:3000/rails/mailers under `Parent Mailer`.

## Follow-up work

This adds the emails themselves, but later on we will obviously need to actually call the functions to send the emails. This will require the work to generate parental consent links to be finished first.

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: dashboard/app/views/parent_mailer/parent_permission_confirmation.html.haml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+%p
+  Dear Parent/Guardian,
```
Comment (mgc1194) [inline dashboard/app/views/parent_mailer/parent_permission_confirmation.html.haml:2]: Curious if these emails are getting translated or if we just provide them in English. Also curious if CPA requires communicating in other languages than English.

Comment (carl-codeorg) [reply dashboard/app/views/parent_mailer/parent_permission_confirmation.html.haml:2]: MVP is English, with Spanish as a nice-to-have

**Comment by wilkie** [comment]
not sure why these (4 similar errors) tests fail for you. I think you have bad luck.

```
ERROR["test_teacher_with_dcdo_who_can_not_see_incubator", "IncubatorHelperTest", 1403.860897430999]
--
1924 | test_teacher_with_dcdo_who_can_not_see_incubator#IncubatorHelperTest (1403.86s)
1925 | Minitest::UnexpectedError:         ActiveRecord::RecordNotUnique: Mysql2::Error: Duplicate entry '80' for key 'PRIMARY'
1926 | config/initializers/mysql_check_index_used.rb:33:in `block (3 levels) in execute'
1927 | config/initializers/mysql_check_index_used.rb:32:in `block (2 levels) in execute'
1928 | config/initializers/mysql_check_index_used.rb:31:in `block in execute'
1929 | config/initializers/mysql_check_index_used.rb:30:in `execute'
1930 | test/helpers/incubator_helper_test.rb:12:in `block in <class:IncubatorHelperTest>'
1931 | test/testing/setup_all_and_teardown_all.rb:36:in `run'
1932 |  
1933 | ==\|
1934 | WARNING: Your progress bar is currently at 6628 out of 6628 and cannot be incremented. In v2.0.0 this will become a ProgressBar::InvalidProgressError.
```

**Comment by wilkie** [comment]
I'd feel like this might not be your fault if it creates this user with a specific id and somehow some other test creates a user that increments to this id first. hmm. https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/test/helpers/incubator_helper_test.rb#L12

File: dashboard/app/mailers/parent_mailer.rb [hunk 2]
```diff
@@ -14,4 +14,22 @@ def parent_email_added_to_student_account(parent_email, student)
...
   end
+
+  # Email for parents of students under 13 to grant permission for their child to have a Code.org account.
+  # The email contains a link that grants permission when clicked.
+  def parent_permission_request(parent_email, permission_url)
+    @permission_url = permission_url
+    mail from: 'Code.org <noreply@code.org>', to: parent_email, subject: I18n.t('parent_mailer.parent_permission_request_subject')
+  end
+
+  # Reminder for parents who haven't granded permission for their child's account.
```
Comment (daynew) [inline dashboard/app/mailers/parent_mailer.rb]:
```suggestion
  # Reminder for parents who haven't granted permission for their child's account.
```
minor misspelling

**Review by daynew — APPROVED** [review]
Looks good! Thank you for documenting how to use the email previews. If you know how to add other language versions for this email, could you create a JIRA task which describes the work and add it to the same CPA epic? That way we can pick it up if we have some more time.

**Pull request merged**